File: php/testcode/benchmark_test_00175.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00175(BenchmarkRequest $req): BenchmarkResponse {
    $data = json_decode($req->bodyStr(), true);
    if ($data === null) {
        return BenchmarkResponse::badRequest('invalid json');
    }
    $name = $data['name'] ?? 'unknown';
    return BenchmarkResponse::ok("hello $name");
}

File: rust/testcode/benchmark_test_00175.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let id = req.param("id");
    match db_query(&id) {
        Ok(r) => super::shared::BenchmarkResponse::ok(&r),
        Err(_e) => super::shared::BenchmarkResponse::error("Internal error. Request ID: req-12345"),
    }
}

fn db_query(_id: &str) -> Result<String, String> { Ok("data".to_string()) }
